File: legacy/_1.10.2/src/main/java/cn/enaium/noexpensive/callback/impl/AnvilSetOutputCallbackImpl.java
package cn.enaium.noexpensive.callback.impl;

import cn.enaium.noexpensive.callback.AnvilSetOutputCallback;
import net.minecraft.client.resource.language.I18n;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.NbtCompound;
import net.minecraft.nbt.NbtList;
import net.minecraft.nbt.NbtString;

/**
 * @author Enaium
 */
public class AnvilSetOutputCallbackImpl implements AnvilSetOutputCallback {
    @Override
    public void interact(ItemStack output, int levelCost, boolean canTake) {
        if (output != null) {
            if (output.getNbt() != null) {
                NbtCompound nbtCompound = output.getSubNbt("display", true);
                nbtCompound.put("Lore", new NbtList());
                nbtCompound.getList("Lore", 9).add(new NbtString(I18n.translate("container.repair.cost", (canTake ? "§a§l" : "§c§l") + levelCost)));
            }
        }
    }
}

File: legacy/_1.8.9/src/main/kotlin/cn/enaium/noexpensive/callback/impl/AnvilSetOutputCallbackImpl.kt
package cn.enaium.noexpensive.callback.impl

import cn.enaium.noexpensive.callback.AnvilSetOutputCallback
import net.minecraft.client.resource.language.I18n
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.ItemStack
import net.minecraft.nbt.NbtList
import net.minecraft.nbt.NbtString


/**
 * @author Enaium
 */
abstract class AnvilSetOutputCallbackImpl : AnvilSetOutputCallback {
    override fun interact(output: ItemStack?, levelCost: Int, canTake: Boolean, player: PlayerEntity) {
        condition(player) && return
        if (output != null) {
            if (output.nbt != null) {
                val nbtCompound = output.getSubNbt("display", true)
                nbtCompound.put("Lore", NbtList())
                nbtCompound.getList("Lore", 9).add(
                    NbtString(
                        I18n.translate(
                            "container.repair.cost",
                            (if (canTake) "§a§l" else "§c§l") + levelCost
                        )
                    )
                )
            }
        }
    }

    abstract fun condition(player: PlayerEntity): Boolean
}